tsdb/executor.go

```diff
@@ -649,29 +649,34 @@ func (e *SelectExecutor) selectorPointToQueryResult(row []interface{}, hasTimeFi
 		if len(e.stmt.Dimensions) > 0 && !hasTimeField {
 			tm = tMin.UTC().Format(time.RFC3339Nano)
 		}
-		row[0] = tm
+		columns[0] = tm
 	}
+
 	for i, c := range columnNames {
 		// skip over time, we already handled that above
 		if i == 0 {
 			continue
 		}
 		if (i == columnIndex && hasTimeField) || (i == columnIndex+1 && !hasTimeField) {
-			row[i] = p.Value
+			// Check to see if we previously processed this column, if so, continue
+			if _, ok := columns[i].(PositionPoint); !ok && columns[i] != nil {
```

Comment:
This is the actual fix.  The problem is that we have to process the entire result for each selector call.  So if you have two selectors (`min` and `max`) in the select statement, it will loop over each row, and each column, for each call, even though I really want it to loop over each row, but ONLY the column it's supposed to deal with.  So the first pass we process the column for `max`, and the second pass we process the column for `min`.  However, due to how we get called (and I couldn't figure out how else to make this work and I'm not happy with it), it would overwrite previously written data.  This check makes sure that we don't process a column (that was a PositionPoint) and re-process it again, this overwriting the correct value we previously wrote.

Now that I understand the problem better it may be that I can find a better approach, but I'm not sure.  This at least fixes the problem for now and doesn't do any more processing than it was already doing, but it does feel like there is an optimization to be had here yet.